File: frontend/website/src/app/models/Car.ts
export class Car {
  private _id:number;
  private _carName:string;
  private _dateAdded:string;


  constructor() {
  }


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get carName(): string {
    return this._carName;
  }

  set carName(value: string) {
    this._carName = value;
  }

  get dateAdded(): string {
    return this._dateAdded;
  }

  set dateAdded(value: string) {
    this._dateAdded = value;
  }
}

File: backend/resource_server/src/main/java/com/example/resource/entities/Car.java
package com.example.resource.entities;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "car")
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "car_name",nullable = false)
    private String carName;

    @Column(name = "date_added",nullable = false)
    private Date dateAdded;

    public Car() {
    }

    public Car(String carName, Date dateAdded) {
        this.carName = carName;
        this.dateAdded = dateAdded;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCarName() {
        return carName;
    }

    public void setCarName(String carName) {
        this.carName = carName;
    }

    public Date getDateAdded() {
        return dateAdded;
    }

    public void setDateAdded(Date dateAdded) {
        this.dateAdded = dateAdded;
    }
}
